### PR TITLE
fixes IngressControllerCaptureHTTPCookie

### PR DESCRIPTION
**Pull request by p0lyn0mial**
as per https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190325-unions.md#go-tags all fields inside a union must be optional.

this PR moves all optional fields to a new struct` IngressControllerCaptureHTTPCookieUnion` and leaves the required in the top level struct

### CODE REVIEW

File: operator/v1/types_ingress.go [hunk 1]
```diff
@@ -660,8 +660,26 @@ const (
 
 // IngressControllerCaptureHTTPCookie describes an HTTP cookie that should be
 // captured.
-// +union
 type IngressControllerCaptureHTTPCookie struct {
+	IngressControllerCaptureHTTPCookieUnion `json:",inline"`
```
Comment (p0lyn0mial) [inline operator/v1/types_ingress.go:664]: callers should start using `IngressControllerCaptureHTTPCookieUnion` instead

Comment (p0lyn0mial) [reply operator/v1/types_ingress.go:664]: new union types should be directly embedded to `IngressControllerCaptureHTTPCookie`

**Comment by sttts** [comment]
/approve
/lgtm

/hold

to give @openshift/sig-network-edge some time to review

**Comment by Miciah** [comment]
Thanks!
/hold cancel
/lgtm

**Pull request merged**